### PR TITLE
Add `ListAccessKeysBulk`

### PR DESCRIPTION
**Pull request by taran-p**
Adds `ListAccessKeysBulk` API to bring `ListAccessKeysLDAPBulk` features to builtin access key listing. Moves `ListAccessKeysLDAP`/`Bulk` to idp-commands.go for consistency

### CODE REVIEW

File: idp-commands.go [hunk 1]
```diff
@@ -437,3 +437,88 @@ func (adm *AdminClient) attachOrDetachPolicyLDAP(ctx context.Context, isAttach b
...
+	queryValues := url.Values{}
+	queryValues.Set("listType", listType)
+	queryValues.Set("userDN", userDN)
+
+	reqData := requestData{
+		relPath:     adminAPIPrefix + "/idp/ldap/list-access-keys",
+		queryValues: queryValues,
+	}
+
+	// Execute GET on /minio/admin/v3/list-service-accounts
```
Comment (shtripat) [inline idp-commands.go:455]: Its `/minio/admin/v3/idp/ldap/list-access-keys` or `/minio/admin/v3/list-service-accounts` ?

Comment (harshavardhana) [inline idp-commands.go:455]:
```suggestion
	// Execute GET on /minio/admin/v3/idp/ldap/list-access-keys
```

File: idp-commands.go [hunk 2]
```diff
@@ -437,3 +437,88 @@ func (adm *AdminClient) attachOrDetachPolicyLDAP(ctx context.Context, isAttach b
...
+
+const (
+	AccessKeyListUsersOnly  = "users-only"
+	AccessKeyListSTSOnly    = "sts-only"
+	AccessKeyListSvcaccOnly = "svcacc-only"
+	AccessKeyListAll        = "all"
+)
+
+// ListAccessKeysLDAPBulk - list service accounts belonging to the given users or all users
+func (adm *AdminClient) ListAccessKeysLDAPBulk(ctx context.Context, users []string, listType string, all bool) (map[string]ListAccessKeysLDAPResp, error) {
```
Comment (harshavardhana) [inline idp-commands.go:486]:
```suggestion
func (adm *AdminClient) ListAccessKeysLDAPBulk(ctx context.Context, users []string, opts ListAccessKeysOpts) (map[string]ListAccessKeysLDAPResp, error) {
```

For future extensibility

File: user-commands.go [hunk 3]
```diff
@@ -597,72 +597,30 @@ func (adm *AdminClient) ListServiceAccounts(ctx context.Context, user string) (L
...
-	AccessKeyListUsersOnly  = "users-only"
-	AccessKeyListSTSOnly    = "sts-only"
-	AccessKeyListSvcaccOnly = "svcacc-only"
-	AccessKeyListAll        = "all"
-)
-
-// ListAccessKeysLDAPBulk - list service accounts belonging to the given users or all users
-func (adm *AdminClient) ListAccessKeysLDAPBulk(ctx context.Context, users []string, listType string, all bool) (map[string]ListAccessKeysLDAPResp, error) {
+// ListAccessKeysBulk - list service accounts belonging to the given users or all users
+func (adm *AdminClient) ListAccessKeysBulk(ctx context.Context, users []string, listType string, all bool) (map[string]ListAccessKeysResp, error) {
```
Comment (harshavardhana) [inline user-commands.go:606]: Add options pattern for future extensibility

**Review by donatello — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
as I have said many times before we do not break published APIs

```
Run TAG="quay.io/minio/minio:9eea7be" make docker
Checking dependencies
Building minio binary to './minio'
# github.com/minio/mc/cmd
Error: ../../../../go/pkg/mod/github.com/minio/mc@v0.0.0-202[4](https://github.com/minio/minio/actions/runs/10862159172/job/30144865752?pr=20436#step:6:5)0909075310-04c5116c9bdf/cmd/idp-ldap-accesskey-list.go:186:84: too many arguments in call to client.ListAccessKeysLDAPBulk
	have (context.Context, []string, string, bool)
	want (context.Context, []string, madmin.ListAccessKeysOpts)
Error: ../../../../go/pkg/mod/github.com/minio/mc@v0.0.0-2024090907[5](https://github.com/minio/minio/actions/runs/10862159172/job/30144865752?pr=20436#step:6:6)310-04c5116c9bdf/cmd/idp-ldap-accesskey-list.go:190:85: too many arguments in call to client.ListAccessKeysLDAPBulk
	have (context.Context, []string, string, bool)
	want (context.Context, []string, madmin.ListAccessKeysOpts)
make: *** [Makefile:1[7](https://github.com/minio/minio/actions/runs/10862159172/job/30144865752?pr=20436#step:6:8)1: build] Error 1
Error: Process completed with exit code 2.
```

https://github.com/minio/minio/actions/runs/10862159172/job/30144865752?pr=20436

We own the mistakes we did in the past

**Comment by taran-p** [comment]
Understood, original PR did not break API, I can revert to that and resubmit if needed.

**Comment by harshavardhana** [comment]
> Understood, original PR did not break API, I can revert to that and resubmit if needed.

Already fixed

**Pull request merged**